 SF: Prime shader cache before boot animation
Primes the shader cache before launching boot animation to avoid
janking when we eventually fall into GLES composition.

diff --git a/services/surfaceflinger/SurfaceFlinger.cpp b/services/surfaceflinger/SurfaceFlinger.cpp
index 4e122da..064e1d5 100644
--- a/services/surfaceflinger/SurfaceFlinger.cpp
+++ b/services/surfaceflinger/SurfaceFlinger.cpp
@@ -531,7 +531,9 @@ void SurfaceFlinger::init() {
     // set initial conditions (e.g. unblank default device)
     initializeDisplays();
 
+#ifndef SHADER_CACHE_OFF
     mRenderEngine->primeCache();
+#endif
 
     // start boot animation
     startBootAnim();
diff --git a/services/surfaceflinger/SurfaceFlinger_hwc1.cpp b/services/surfaceflinger/SurfaceFlinger_hwc1.cpp
index 4e122da..064e1d5 100644
--- a/services/surfaceflinger/SurfaceFlinger_hwc1.cpp
+++ b/services/surfaceflinger/SurfaceFlinger_hwc1.cpp
@@ -579,7 +579,9 @@ void SurfaceFlinger::init() {
     // set initial conditions (e.g. unblank default device)
     initializeDisplays();
 
+#ifndef SHADER_CACHE_OFF
     mRenderEngine->primeCache();
+#endif
 
     // start boot animation
     startBootAnim();
